app/controllers/sitemap_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SitemapController < ApplicationController
+  def index
+    all_urls = [high_voltage_urls, route_urls, service_specific_information_urls, tutorial_urls]
+
+    @url_set = Set.new
+    all_urls.each do |urls|
+      @url_set.merge(urls)
+    end
+  end
+
+  def high_voltage_urls
+    HighVoltage.page_ids.map { |page| page_url(page) if page != 'homepage' }.compact
+  end
+
+  def route_urls
+    routes = Rails.application.routes.routes
+    paths = routes.map { |route| route.path.spec.to_s[%r{^\/([^\(:*]+)}, 1] }.compact
+    paths.map { |path| page_url(path.chomp('/')) unless path.match?(/^(rails|assets|sitemap|homepage|\d)/) }.compact
```

Comment:
This set of path prefixes to exclude will probably change less than the set of path prefixes to include. However, given the complexity of this function I'm also fine with just scrapping it and hardcoding the paths to include. Thoughts?

Reply:
+1 on your line of thinking but _good heavens_ that's some codez.